### ISSUE TITLE
Dark Theme

### ISSUE THREAD

**Issue by Kalibr8** [ayu-theme/ayu-vim#57]
let ayucolor="dark" not working. Theme is stuck on default pallete

**Comment by stsewd** [comment]
Please share your minimal init.vim/vimrc file

**Comment by Kalibr8** [comment]
`set termguicolors

filetype off
filetype plugin on
filetype plugin indent on


"========================================================================================"
"-----------------------------------Plugins (Start)--------------------------------------"
"========================================================================================"


call plug#begin('~/.local/share/nvim/plugged')



Plug 'itchyny/lightline.vim'

Plug 'ayu-theme/ayu-vim'


Plug 'kassio/neoterm'
Plug 'bfredl/nvim-ipy'
Plug 'vim-utils/vim-man'

Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }

" (Optional) Multi-entry selection UI.
Plug 'junegunn/fzf'
let g:LanguageClient_serverCommands = {
  \ 'cpp': ['clangd'],
  \ }


Plug 'neomake/neomake'
Plug 'tracyone/neomake-multiprocess'
Plug 'airblade/vim-gitgutter'
Plug 'artur-shaik/vim-javacomplete2'
Plug 'tpope/vim-vinegar'
Plug 'tpope/vim-obsession'
Plug 'tpope/vim-fugitive'
Plug 'junegunn/gv.vim'
Plug 'itchyny/vim-gitbranch'
Plug 'majutsushi/tagbar'
"Plug 'zchee/deoplete-jedi'
Plug 'szymonmaszke/vimpyter'
Plug 'tpope/vim-dadbod'
Plug 'Shougo/denite.nvim'
Plug 'sheerun/vim-polyglot'
Plug 'CoatiSoftware/vim-sourcetrail'
Plug 'rhysd/wandbox-vim'
Plug 'universal-ctags/ctags'
Plug 'ludovicchabant/vim-gutentags'
Plug 'dbakker/vim-projectroot'
"Plug 'starcraftman/vim-eclim'
Plug 'numirias/semshi', {'do': ':UpdateRemotePlugins'}
Plug 'jaxbot/browserlink.vim'
Plug 'scrooloose/nerdtree'
Plug 'Xuyuanp/nerdtree-git-plugin'
Plug 'mhinz/vim-startify' 
Plug 'tiagofumo/vim-nerdtree-syntax-highlight'
Plug 'luochen1990/rainbow'
Plug 'sheerun/vim-polyglot'
Plug 'dense-analysis/ale'
Plug 'maximbaz/lightline-ale'
Plug 'neoclide/coc.nvim', {'branch': 'release'}
Plug 'jiangmiao/auto-pairs'
Plug 'tpope/vim-surround'
Plug 'justinmk/vim-dirvish'
Plug 'tpope/vim-commentary'
Plug 'vim-vdebug/vdebug'
Plug 'JoshMcguigan/estream', { 'do': 'bash install.sh v0.1.1' }
Plug 'skywind3000/asyncrun.vim'
Plug 'mattn/emmet-vim'
Plug 'tpope/vim-ragtag'
Plug 'tpope/vim-repeat'
Plug 'glts/vim-magnum'
Plug 'glts/vim-radical'
Plug 'mengelbrecht/lightline-bufferline'
Plug 'voldikss/vim-floaterm'
Plug 'ryanoasis/vim-devicons'

call plug#end()

"========================================================================================"
"-----------------------------------Plugins (End)----------------------------------------"
"========================================================================================"

let ayucolor="dark"
colorscheme ayu`

**Comment by aasril** [comment]
You may also need to run `:colorscheme ayu` after you run `:let ayucolor="dark"` from the command line. Because I do.

**Comment by panosl** [comment]
> You may also need to run `:colorscheme ayu` after you run `:let ayucolor="dark"` from the command line. Because I do.

Doesn't it make sense to register them as 3 themes? colo ayu-dark, ayu-mirage, ayu-light?

Having to execute `:let ayucolor='dark'` and then `:colo ayu`, seems kinda cumbersome.

**Comment by iusmac** [comment]
> > You may also need to run `:colorscheme ayu` after you run `:let ayucolor="dark"` from the command line. Because I do.
> 
> Doesn't it make sense to register them as 3 themes? colo ayu-dark, ayu-mirage, ayu-light?
> 
> Having to execute `:let ayucolor='dark'` and then `:colo ayu`, seems kinda cumbersome.

Just take a look inside the `colors/ayu.vim` file and you will see that it makes sense to define a variable instead of creating more *.vim files and violating the "do not repeat yourself" rule. I use the help of VimL syntax to create a one-liner and to make them visually dependent on each other like `let ayucolor='dark' | colo auy`. I've done this once and see no problems. The way the auy theme is designed is a really good compromise for both users and contributors. IMHO.

**Comment by panosl** [comment]
> > > You may also need to run `:colorscheme ayu` after you run `:let ayucolor="dark"` from the command line. Because I do.
> > 
> > 
> > Doesn't it make sense to register them as 3 themes? colo ayu-dark, ayu-mirage, ayu-light?
> > Having to execute `:let ayucolor='dark'` and then `:colo ayu`, seems kinda cumbersome.
> 
> Just take a look inside the `colors/ayu.vim` file and you will see that it makes sense to define a variable instead of creating more *.vim files and violating the "do not repeat yourself" rule. I use the help of VimL syntax to create a one-liner and to make them visually dependent on each other like `let ayucolor='dark' | colo auy`. I've done this once and see no problems. The way the auy theme is designed is a really good compromise for both users and contributors. IMHO.

You're not wrong, but I don't like that I have to change mindset when I use that theme compared to others.

i.e. I use `:colo <theme>` for everything, but then I need to `:let ayucolor='<something>'` and then set the theme.

Maybe #54 solves some these issues.

**Comment by themysteryboy** [comment]
I've tried, you only need to change the "let ayucolor='dark' to 'let ayucolor='dark' | color ayu' and it the color will be what you want it to be

**Comment by Kalibr8** [comment]
Solved it on my end. Just had to set the colorscheme and background, as well as termguicolors, _after_ the plugins. Also, set
 the background to dark before setting ayucolor to dark and make sure all terminal background settings are dark after setting colorscheme, i.e

call plug#end()                "for vim-plug

set termguicolors
set background=dark
let ayucolor="dark"
colorscheme ayu  
hi Normal ctermbg=16 guibg=#000000
hi LineNr ctermbg=16 guibg=#000000
highlight NonText guibg=black
highlight NonText guifg=black
